src/Tools/Microsoft.dotnet-openapi/src/Commands/AddURLCommand.cs

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.Extensions.CommandLineUtils;
+using Microsoft.Extensions.Tools.Internal;
+
+namespace Microsoft.DotNet.OpenApi.Commands
+{
+    internal class AddURLCommand : BaseCommand
+    {
+        private const string CommandName = "url";
+        private const string DefaultOpenAPIDir = "openapi";
+        private const string DefaultOpenAPIFile = "openapi.json";
+
+        private const string OutputFileName = "--output-file";
+
+        public AddURLCommand(AddCommand parent)
+            : base(parent, CommandName)
+        {
+            _outputFileOption = Option(OutputFileName, "The destination to download the remote Open API file to.", CommandOptionType.SingleValue);
+            _sourceFileArg = Argument(SourceProjectArgName, $"The openapi file to add. This must be a URL to a remote openapi file.", multipleValues: true);
+        }
+
+        internal readonly CommandOption _outputFileOption;
+
+        internal readonly CommandArgument _sourceFileArg;
+
+        protected override async Task<int> ExecuteCoreAsync()
+        {
+            var projectFilePath = ResolveProjectFile(ProjectFileOption);
+
+            var sourceFile = Ensure.NotNullOrEmpty(_sourceFileArg.Value, SourceProjectArgName);
+
+            string outputFile;
+            if (_outputFileOption.HasValue())
+            {
+                outputFile = _outputFileOption.Value();
+            }
+            else
+            {
+                outputFile = Path.Combine(DefaultOpenAPIDir, DefaultOpenAPIFile);
```

Comment:
If you'd like that functionality I say file an issue, but it's definitely not something I want to complicate this PR with.

Reply:
Does the user get feedback to say where the file gets written to? Is that something we could print?

Reply:
https://github.com/aspnet/AspNetCore/pull/12810/files#diff-a00ae2e84684f653a9e023f4511cfc37R368

Reply:
That's [not the case](https://github.com/aspnet/AspNetCore/pull/12810/files#diff-a00ae2e84684f653a9e023f4511cfc37R365). When adding a new Reference we check to see if the destination file already exists, and fails (loudly) if it does. I added a test case for this.

Reply:
I'm glad the failure isn't actually silent.

But, why is it an error when users try to add references to multiple documents from the Internet? The default should be to allow that and to avoid collisions between the downloaded files, not to create error cases.